lib/rubyXL/objects/sheet_data.rb

```diff
@@ -107,7 +107,7 @@ def value(args = {})
       when RubyXL::DataType::INLINE_STRING then is.to_s
       when RubyXL::DataType::RAW_STRING    then raw_value
       else
-        if is_date? then workbook.num_to_date(r.to_f)
+        if r.is_a?(String) && is_date? then workbook.num_to_date(r.to_f)
```

Comment:
This needs to be carefully considered. Yes, this expression would be correct for the files freshly read from `XLSX`. However, if you insert `Date` into a cell using `change_contents`, it will be stored as `Float` and this line will be bypassed, which is not the desirable behavior.

Reply:
If I’m not mistaken, any Float will return nil against the regex, meaning that it already will not respond as a date.
Although this may not be the desirable behavior, it is the current behavior.